Source/src/importers/MeshImporter.cpp

```diff
@@ -138,17 +152,42 @@ Hachiko::Resource* Hachiko::MeshImporter::Load(UID id)
     else
     {
         mesh->tangents = nullptr;
-    }   
+    }
+    if (sizes[static_cast<int>(ResourceMesh::Buffers::BONES_INDICES)] > 0)
+    {
+        mesh->src_bone_indices = std::make_unique<unsigned[]>(sizes[static_cast<int>(ResourceMesh::Buffers::BONES_INDICES)]);
+        size_bytes = sizeof(unsigned) * sizes[static_cast<int>(ResourceMesh::Buffers::BONES_INDICES)];
+        memcpy(mesh->src_bone_indices.get(), cursor, size_bytes);
+        cursor += size_bytes;
+    }
+    else
+    {
+        mesh->src_bone_indices = nullptr;
+    }
+    if (sizes[static_cast<int>(ResourceMesh::Buffers::BONES_WEIGHTS)] > 0)
+    {
+        mesh->src_bone_weights = std::make_unique<float4[]>(sizes[static_cast<int>(ResourceMesh::Buffers::BONES_WEIGHTS)]);
+        size_bytes = sizeof(float4) * sizes[static_cast<int>(ResourceMesh::Buffers::BONES_WEIGHTS)];
+        memcpy(mesh->src_bone_weights.get(), cursor, size_bytes);
+        cursor += size_bytes;
+    }
+    else
+    {
+        mesh->src_bone_weights = nullptr;
+    } 
```

Comment:
I believe you follow what we had previously here. This is not a comment for you, but rather a note to the future us, we can delete all this `if else` hell, and just assign `nullptr` or `0` to everything at start and just leave the `if`s.